gammapy/modeling/models/core.py

```diff
@@ -791,6 +791,30 @@ def reassign(self, dataset_name, new_dataset_name):
         models = [m.reassign(dataset_name, new_dataset_name) for m in self]
         return self.__class__(models)
 
+    def to_template_spatial_model(self, geom, spectral_model=None, name=None):
```

Comment:
I think here we should take a decision:
- Either name it `.to_template_spatial_model()` and return a `TemplateSpatialModel`
- Or rename to `.to_template_sky_model()`
I think I prefer the first option, because the latter basically duplicates the API of creating a `SkyModel()` by requiring the `spectral_model` and `name` keywords. But I can see the additional convenience, that saves a few lines of code for the standard use case...